### PR TITLE
Warn instead of failing on empty or invalid alt-text

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Tentative solution to alt text failures due to an empty string (#4501).

I don't know what is our preferred solution to handle other kinds of `ParserError`. A more conservative alternative would be to strip the string and check if it's empty, instead.

cc @melissawm, @cnstt

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by melissawm** [comment]
This looks reasonable to me - @jni did you want to be more strict about missing alt-text?

**Comment by cnstt** [comment]
So the where would the warnings will be displayed? During the building of the website I suppose?
It would be good to be able to see/collect those errors to be able to correct them! Or maybe even better if there are already tests existing for the documentation, to add one that tests if 'alt text' is empty. As such we would never forget to add them anymore, when new documentation is written!

File: napari/utils/notebook_display.py [hunk 1]
```diff
@@ -86,7 +87,13 @@ def _clean_alt_text(self, alt_text):
                 str(alt_text)
             )  # cleaner won't recognize unescaped script tags
             cleaner = Cleaner()
-            doc = document_fromstring(alt_text)
+            try:
+                doc = document_fromstring(alt_text)
+            except ParserError:
+                alt_text = ""
+                warn(
+                    'The provided alt text does not constitute valid html, so it was discarded.'
```
Comment (Carreau) [inline napari/utils/notebook_display.py]: Maybe we want to warn with a specific warning class, so that it can be filtered/turned into error. If we don't I believe the default class will be USerWarning. I guess we could even warng with `ParseError`... even if that may be weird.

File: napari/utils/notebook_display.py [hunk 2]
```diff
@@ -86,7 +87,13 @@ def _clean_alt_text(self, alt_text):
...
+                )
```
Comment (Czaki) [inline napari/utils/notebook_display.py:97]:
```suggestion
		stacklevel=3,
                )
```
To raise a warning in the place where the constructor of `NotebookScreenshot` is called. 

Or if public API just uses this class please increase this number. 

It may be worth adding `sctacklevel` to the previous warning about the lack of lxml

**Review by Czaki — DISMISSED** [review]

File: napari/utils/notebook_display.py (hunk 2 above)
Comment (brisvag) [reply napari/utils/notebook_display.py:97]: I see, I never quite understood the stacklevel argument (I always reused warnings from the same code area :P).

**Comment by isabela-pf** [comment]
I'm not entirely familiar with this problem or discussion, but this approach sounds similar to the behavior that got approval on [the discussion for adding alt text as an argument for nbscreenshot](https://github.com/napari/napari/pull/3825). 

In short, I think this matches what we already agreed upon. Thanks for opening the PR, @brisvag!

File: napari/utils/notebook_display.py [hunk 3]
```diff
@@ -85,9 +86,20 @@ def _clean_alt_text(self, alt_text):
...
+            cleaner = Cleaner()
+            try:
+                doc = document_fromstring(alt_text)
+            except ParserError:
+                alt_text = ""
+                warn(
+                    'The provided alt text does not constitute valid html, so it was discarded.',
+                    stacklevel=3,
+                )
+            alt_text = cleaner.clean_html(doc).text_content()
```
Comment (jni) [inline napari/utils/notebook_display.py]: @brisvag Currently this is an UnboundLocalError, because `doc` fails to be assigned in the try clause. Reproducer:

```python
import napari
from napari.utils.notebook_display import NotebookScreenshot

viewer = napari.Viewer()
nbscr = NotebookScreenshot(viewer, alt_text=" ")
```

So this needs to be fixed and then a test added for good measure. 😂

**Review by jni — DISMISSED** [review]
The try/except logic is currently broken because `doc` is only assigned if the try succeeds, but gets used regardless in line 98 `alt_text = cleaner.clean_html(doc).text_content()`.

File: napari/utils/notebook_display.py (hunk 3 above)
Comment (brisvag) [reply napari/utils/notebook_display.py]: oops, things got entangled in refactoring. Should work now!

**Review by Czaki — COMMENTED** [review]
Could you add a test for it?

**Review by Czaki — APPROVED** [review]
